targets/tizenrt-artik05x/configs/jerryscript/Make.defs

```diff
@@ -1,5 +1,37 @@
+###########################################################################
+# Copyright JS Foundation and other contributors, http://js.foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+###########################################################################
+#
+# Copyright 2017 Samsung Electronics All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
```

Comment:
@glistening Well, that's not exactly what I was trying saying. This commit tries to add an Apache 2.0 license header with  a `Copyright 2017 Samsung Electronics All Rights Reserved.` notice. But that's explicitly prohibited by the contribution guidelines (as linked and quoted in my above comment). The Apache 2.0 license header must be accompanied by the `Copyright JS Foundation and other contributors, http://js.foundation` project-wide notice:

> Adding copyright notices other than the project-wide notice ("Copyright JS Foundation and other contributors, http://js.foundation") is not permitted.

As for BSD or BSD-like licenses (e.g., the NuttX license): I did not want to suggest its removal (I think that would explicitly go against the provisions of BSD3.) Such a compatible license block (together with its copyright notice) can be kept below the Apache 2.0 license block (that's why I referred to `jerry-libm` and `ecma-helpers-errol.c`, which also contain an additional - although not BSD but other Apache 2.0 compatible - license block). Fortunately, this is permitted by our guidelines:

> The only exception is adding third-party code which requires copyright notices to be preserved.

As for the `third-party/valgrind` library: I don't exactly know the history of that part. It's been there for a while and has not been touched since it's addition in 2014. I don't really know why it was not touched when the project was transferred to JS Foundation. However, I do think that any new code/contribution should follow the current guidelines.